python/ray/experimental/gpu_object_manager/gpu_object_store.py

```diff
@@ -55,29 +61,59 @@ def __ray_send__(self, communicator_name: str, obj_id: str, dst_rank: int):
             raise ValueError(
                 f"tensor device {tensor.device} does not match device {device}"
             )
-        collective.send(tensor, dst_rank, group_name=communicator_name)
+        if backend != Backend.NIXL:
+            collective.send(tensor, dst_rank, group_name=communicator_name)
 
 
 def __ray_recv__(
     self,
     communicator_name: str,
     obj_id: str,
     src_rank: int,
-    tensor_meta: List[Tuple["torch.Size", "torch.dtype"]],
+    tensor_meta: Tuple[
+        List[Tuple["torch.Size", "torch.dtype"]], Optional[bytes], Optional[bytes]
+    ],
 ):
     """Helper function that runs on the dst actor to receive tensors from the src actor."""
     from ray._private.worker import global_worker
 
-    backend = collective.get_group_handle(communicator_name).backend()
+    if communicator_name != "nixl":
```

Comment:
Nit: reverse so we do `if communicator_name == "nixl"` first.

Reply:
Updated